broker-daemon/block-order-worker/index.spec.js

```diff
@@ -1234,25 +1234,71 @@ describe('BlockOrderWorker', () => {
       let failBlockOrderStub
 
       beforeEach(() => {
+        fill = {
+          id: 'anotherId',
+          once: onceStub,
+          removeAllListeners: removeAllListenersStub,
+          fill: {
+            fillAmount: '100',
+            error: {
+              metadata: new Map([['invalidOrderStatus', ['true']]])
+            }
+          }
+        }
+        FillStateMachine.create.resolves(fill)
         failBlockOrderStub = sinon.stub().resolves(true)
         worker.failBlockOrder = failBlockOrderStub
       })
 
-      beforeEach(async () => {
+      it('registers an event on an order for order rejection', async () => {
         await worker._fillOrders(blockOrder, orders, targetDepth)
+        expect(onceStub).to.have.been.calledWith('reject', sinon.match.func)
       })
 
-      it('registers an event on an order for order rejection', async () => {
-        expect(onceStub).to.have.been.calledWith('reject', sinon.match.func)
+      it('removes all listeners if the call is rejected', async () => {
+        await worker._fillOrders(blockOrder, orders, targetDepth)
+        worker.workBlockOrder = sinon.stub().resolves(true)
+        await onceStub.args[1][1]()
+        expect(fill.removeAllListeners).to.have.been.calledOnce()
       })
 
-      it('fails a block order if the call is rejected', async () => {
+      it('reworks a blockOrder if the order was already filled', async () => {
```

Comment:
I would call this `reworks a blockOrder if the order is not in a state to be filled` since it could be in another state for lots of reasons (cancelled, not enough funds in channel, etc)